fix: Replaced match with HashMap for faster evdevmapping

nots1dd opened the pull request:
Solves #4

Added lazy_static to Cargo.toml

1. Optimized evdev mapping insertion times with HashMap
2. Loaded the mapping lazily

Tested the above changes with:

```bash
cargo test
```

nots1dd committed:
fix: Replaced match with HashMap for faster evdevmapping

Added lazy_static to Cargo.toml

1. Optimized evdev mapping insertion times with HashMap
2. Loaded the mapping lazily

lavafroth commented:
Hi, could we use phf for this? It would build the lookup table at compile time, which is especially useful for us since we already know all the keys in the map.

Perhaps check out https://docs.rs/phf

nots1dd commented:
> Hi, could we use phf for this? It would build the lookup table at compile time, which is especially useful for us since we already know all the keys in the map. Perhaps check out https://docs.rs/phf

Ah yes that makes sense, I'll take a look at it and modify the PR then

nots1dd committed:
fix: Use phf for loading evdev map at compile time

Removed lazy_static from Cargo.toml.
Added phf and phf macros to Cargo.toml

1. Replaced `HashMap` with `phf::Map` for loading the static evdev
mapping at compile time

nots1dd commented:
> Hi, could we use phf for this? It would build the lookup table at compile time, which is especially useful for us since we already know all the keys in the map. Perhaps check out https://docs.rs/phf

I have made the requested changes, the API was quite simple to work with. Also I didnt add function keys till 24 (so f1-f12) since the test was passing without them, is this fine?

nots1dd committed:
chore: add remaining mappings

nots1dd committed:
chore: change static key map name

nots1dd commented:
> > Hi, could we use phf for this? It would build the lookup table at compile time, which is especially useful for us since we already know all the keys in the map. Perhaps check out https://docs.rs/phf
> 
> I have made the requested changes, the API was quite simple to work with. Also I didnt add function keys till 24 (so f1-f12) since the test was passing without them, is this fine?

Never mind, I still have added all the previous keymaps that existed to not cause any trouble in the future.

lavafroth commented:
Thank you for contributing!
